### ISSUE TITLE
NNBD: Dart does not throw compile error when subtyping brings together incompatible member signatures for the method with parameter overriden in the opted in code. 

### ISSUE THREAD

**Issue by iarkh** [dart-lang/sdk#41529]
Dart VM version: 2.8.0-dev.20.0 (dev) (Fri Apr 3 10:19:55 2020 +0200) on "windows_x64"

This issue is similar with #40414, but describes slightly different case.

The following code source brings together otherwise incompatible member signatures in the legacy code, then opted in method inherits them:

testlib_in.dart:
``` dart
class NULLABLE {
  void test_int (int? i) {}
}

class NONNULLABLE {
  void test_int (int i) {}
}
```
testlib_out.dart:
``` dart
// @dart=2.6
import "testlib_in.dart";
class A extends NULLABLE implements NONNULLABLE {}
```
test.dart:
``` dart
import "testlib_out.dart";

class B extends A {}

class C extends A {
  void test_int (int i) {}
}

main() {
  A().test_int(1);
  B().test_int(1);
  C().test_int(1);
}
```
It throws an expected compile error with analyzer and passes without any errors or warnings with dart.
Sample output is:
> $ dartanalyzer --enable-experiment=non-nullable test.dart
> Analyzing test.dart...
>   error - 'C.test_int' ('void Function(int)') isn't a valid override of 'NULLABLE.test_int' ('void Function(int?)'). - test.dart:6:8 - invalid_override
> 1 error found.
> $ dart --enable-experiment=non-nullable test.dart
> 
> $

**Comment by eernstg** [comment]
This is actually not an error. The interface of `A` consists exclusively of [legacy member signatures](https://github.com/dart-lang/language/blob/master/accepted/future-releases/nnbd/feature-specification.md#classes-defined-in-legacy-libraries-as-seen-from-opted-in-libraries), so it has `void* test_int(int*)`, and the override relationship between that signature and `void test_int(int)` is correct.

It is also important to note that the override relationship is always checked using the current class (here: `C`) and each of its direct superinterfaces (there is one: `A`), and each interface contains all the member signatures, declared by the corresponding class or inherited via some superinterface. So we don't ever search the entire superinterface graph from `C` and find `void test_int(int?)` in `NULLABLE`, instead we summarize all the information at each step. This means that all the information from `NULLABLE` is carried down to be part of the interface of `A`. Then, when we erase all member signatures in the interface of `A`, we'll just see `void* test_int(int*)`.

The common front end already implements these rules, and I expect that the analyzer will do it soon.

**Comment by scheglov** [comment]
This was fixed in https://dart-review.googlesource.com/c/sdk/+/142760, landed as 7e4673f497f978b4e2c72939b6786f3e45c25265. This test does not report any errors now.
